File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/recipes/InteractionWithCustomName.kt
package examples.kotlin.recipes

import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import com.ing.baker.recipe.kotlindsl.recipe
import examples.kotlin.interactions.ShipOrder

@ExperimentalDsl
object InteractionWithCustomName {
    val recipe = recipe("example") {
        interaction<ShipOrder> {
            name = "ship-order"
        }
    }
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/InteractionWithCustomName.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.Recipe
import examples.scala.interactions.ShipOrder

object InteractionWithCustomName {
  val recipe: Recipe = Recipe("example")
    .withInteraction(
      ShipOrder.interaction
        .withName("ship-order")
    )
}